homeassistant/components/mobile_app/notify.py

```diff
@@ -124,81 +131,144 @@ async def async_send_message(self, message="", **kwargs):
 
-            if "message" in result:
-                if message[-1] not in [".", "?", "!"]:
-                    message += "."
-                message += " This message is generated externally to Home Assistant."
+        if response.status in (
+            HTTPStatus.OK,
+            HTTPStatus.CREATED,
+            HTTPStatus.ACCEPTED,
+        ):
+            log_rate_limits(hass, registration[ATTR_DEVICE_NAME], result)
+            return
+
+        fallback_error = result.get("errorMessage", "Unknown error")
+        fallback_message = (
+            f"Internal server error, please try again later: {fallback_error}"
+        )
+        message = result.get("message", fallback_message)
+
+        if "message" in result:
+            if message[-1] not in [".", "?", "!"]:
+                message += "."
+            message += " This message is generated externally to Home Assistant."
+
+        if response.status == HTTPStatus.TOO_MANY_REQUESTS:
+            _LOGGER.warning(message)
+            log_rate_limits(
+                hass, registration[ATTR_DEVICE_NAME], result, logging.WARNING
+            )
+        else:
+            _LOGGER.error(message)
 
-            if response.status == HTTPStatus.TOO_MANY_REQUESTS:
-                _LOGGER.warning(message)
-                log_rate_limits(
-                    self.hass, registration[ATTR_DEVICE_NAME], result, logging.WARNING
-                )
-            else:
-                _LOGGER.error(message)
+    except TimeoutError:
+        _LOGGER.error("Timeout sending notification to %s", push_url)
+    except aiohttp.ClientError as err:
+        _LOGGER.error("Error sending notification to %s: %r", push_url, err)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the demo entity platform."""
+    registration = config_entry.data
+
+    if not supports_push(hass, registration[CONF_WEBHOOK_ID]):
+        _LOGGER.debug("Target does not support push")
+        return
+
+    async_add_entities([MobileAppNotifyEntity(dict(registration))])
+
+
+class MobileAppNotifyEntity(NotifyEntity):
+    """Implement demo notification platform."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_supported_features = NotifyEntityFeature.TITLE
+
+    def __init__(
+        self,
+        registration: dict,
+    ) -> None:
+        """Initialize the Demo button entity."""
+        self._registration = registration
+        self._attr_unique_id = registration[ATTR_DEVICE_ID]
+        self._attr_device_info = device_info(registration)
+
+    async def async_send_message(self, message: str, title: str | None = None) -> None:
+        """Send a message to a user."""
+
+        data = {ATTR_MESSAGE: message}
+
+        # Remove default title from notifications.
+        if title is not None and title != ATTR_TITLE_DEFAULT:
+            data[ATTR_TITLE] = title
+
```

Comment:
Do we want to move forward with this dispite lacking ATTR_DATA support?

Reply:
Not sure how that is used. As we should keep the old service for the deprecation period we could deprecate the use of it?

Reply:
It has quite a lot of features in data: https://companion.home-assistant.io/docs/notifications/notifications-basic/

So i dont think we can deprecate the old service untill we replicate some of that.

Reply:
Maybe the data could ve namespaced by some domain key.

```yaml
data:
  mobile_app:
     bla_data
  other_integration:
     other_data
```

That way it would work with group too.

Reply:
Right, in that case it seems not we can deprecate the old service.
We could consider to implement both, the entity and legacy service?
We could deprecate the old service as soon as the new entity service is ready to replace it.

Reply:
Another option is that the new service allows to set up some templated notifiy entities via config flow. In that way the user can still send the message to the entity, but the formatting might be no longer part of the message.

Reply:
We likely need config sub entries in place for that, since the data to add would be quite custom. But the idea seems quite sound.
That said, i have never ever used any of these advanced features so not sure how they are used :)